libraries/common/list.effekt

```diff
@@ -167,9 +167,52 @@ def foreachIndex[A](list: List[A]){ f: (Int, A) {Control} => Unit }: Unit = {
 /// Map a function `f` over elements in a given list.
 ///
 /// O(N)
-def map[A, B](l: List[A]) { f: A => B } : List[B] = {
-  var acc = Nil[B]()
-  l.foreach { el => acc = Cons(f(el), acc) }
+def map[A, B](l: List[A]) { f: A => B }: List[B] = {
+  var acc: List[B] = Nil()
+  l.foreach { el =>
+    acc = Cons(f(el), acc)
+  }
+  acc.reverse
+}
```

Comment:
This is exactly `foldRight(Nil()) { (x, acc) => Cons(x, acc) }`, no?

Reply:
Agreed, but I think our focus is on showing more "imperative" versions :)
Note that I haven't changed the style `map` is written in -- that's the same as it always was.

(And the version with a mutable variable was quicker last time I checked!)

Reply:
I noted that, but I have never checked the implementation before. I don't find "this is the same it has always been" a very compelling argument, though.